### ISSUE TITLE
 Provider com.sksamuel.hoplite.decoder.KerberosPrincipalDecoder not found

### ISSUE THREAD

**Issue by davidmoshal** [sksamuel/hoplite#71]
Project runs fine in IDE, but when built into a jar file with ShadowJar, fails with:
```
Exception in thread "main" java.util.ServiceConfigurationError: com.sksamuel.hoplite.decoder.Decoder: Provider com.sksamuel.hoplite.decoder.KerberosPrincipalDecoder not found
```

The KerberosPrincipalDecoder.class is not in the jar file.
Curious why this is a problem when run from the jar file, and not the IDE?
Any thoughts, what additional info would be helpful?

Thanks
David

**Comment by davidmoshal** [comment]
problem seemed to resolve when switching this:
```kotlin
tasks.withType<ShadowJar>() {
   ...
}
```
to this:
```kotlin
tasks.withType<Jar>() {
   ...
}
```
The various classes are all in the jar file,
so, probably a problem with ShadowJar, not hoplite.

Separately: the jar file has an 18.5 Mb /arrow class directory.
Any way to make Arrow optional?

**Comment by sksamuel** [comment]
The project is built on arrow so it's needed but that does seem a lot. Let
me look at narrowing down the arrow scope.

**Comment by davidmoshal** [comment]
Great, thanks for the awesome project, very useful.

**Comment by sksamuel** [comment]
I've released 1.2.0 where I've removed arrow as a core dependency. So if you were to try this version, the jar size should be massively smaller.
